Source/Tests/LinuxSyscalls/x64/Memory.cpp

```diff
@@ -1,19 +1,57 @@
 #include "Tests/LinuxSyscalls/Syscalls.h"
 #include "Tests/LinuxSyscalls/x64/Syscalls.h"
 #include <FEXCore/Core/Context.h>
+#include <FEXCore/Debug/InternalThreadState.h>
 
 #include <sys/mman.h>
 #include <sys/shm.h>
+#include <map>
+#include <unistd.h>
+
+#include <FEXCore/Core/Context.h>
+#include <FEXCore/Config/Config.h>
+#include <fstream>
+
+std::string get_fdpath(int fd)
+{
+    std::vector<char> buf(400);
+    ssize_t len;
+
+    std::string fdToName = "/proc/self/fd/" + std::to_string(fd);
+
+    do
+    {
+        buf.resize(buf.size() + 100);
+        len = ::readlink(fdToName.c_str(), &(buf[0]), buf.size());
```

Comment:
`std::filesystem::absolute()`

Reply:
I just want to resolve the softlink there, I don't think ::abosulte() does that

Reply:
whoops, you want `std::filesystem::canonical()` instead

Reply:
I don't think readlink guarantees a canonical path, it only follows one simlink